packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -260,14 +260,14 @@ class AOTSnapshotter {
 
     const String embedBitcodeArg = '-fembed-bitcode';
     final String assemblyO = _fileSystem.path.join(outputPath, 'snapshot_assembly.o');
-    List<String>? isysrootArgs;
     if (sdkRoot != null) {
-      isysrootArgs = <String>['-isysroot', sdkRoot];
+      commonBuildOptions.addAll(<String>[
+        '-isysroot', sdkRoot,
+      ]);
     }
 
     final RunResult compileResult = await _xcode.cc(<String>[
-      '-arch', targetArch,
-      if (isysrootArgs != null) ...isysrootArgs,
+      ...commonBuildOptions,
```

Comment:
`commonBuildOptions` is `-arch`, `-miphoneos-version-min`, and `-isysroot`.

Reply:
nit: hoist this up to where commonBuildOptions is declared?

```
    final List<String> commonBuildOptions = <String>[
      '-arch', targetArch,
      if (isIOS)
        // When the minimum version is updated, remember to update
        // template MinimumOSVersion.
        // https://github.com/flutter/flutter/pull/62902
        '-miphoneos-version-min=8.0',
        if (sdkRoot != null) '-isysroot=$sdkroot',
    ];
```

Reply:
Ha oh yeah that's better...